watertap/unit_models/steam_heater.py

```diff
@@ -0,0 +1,116 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+from pyomo.environ import Var
+
+from idaes.core import (
+    declare_process_block_class,
+)
+from idaes.models.unit_models.heat_exchanger import HeatExchangerData
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.logger as idaeslog
+
+
+_log = idaeslog.getLogger(__name__)
+
+
+__author__ = "Elmira Shamlou"
+
+
+@declare_process_block_class(
+    "FWHCondensing0D",
+    doc="""Feedwater Heater Condensing Section
+The feedwater heater condensing section model is a normal 0D heat exchanger
+model with an added constraint to calculate the steam flow such that the outlet
+of shell is a saturated liquid.""",
+)
+class FWHCondensing0DData(HeatExchangerData):
+    config = HeatExchangerData.CONFIG()
+
+    def build(self):
+        super().build()
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.hot_side.config.property_package.component_list,
+            doc="Mass balance",
+        )
+        def outlet_liquid_mass_balance(b, t, j):
+            b.hot_side.properties_out[t].flow_mass_phase_comp["Vap", j].fix(0)
+            return (
+                b.hot_side.properties_in[t].flow_mass_phase_comp["Vap", j]
+                + b.hot_side.properties_in[t].flow_mass_phase_comp["Liq", j]
+                == b.hot_side.properties_out[t].flow_mass_phase_comp["Liq", j]
+            )
+
+        units_meta = (
+            self.config.hot_side.property_package.get_metadata().get_derived_units
+        )
+
+        self.pressure_diff = Var(
+            self.flowsheet().time, initialize=0, units=units_meta("pressure")
+        )
+
+        self.pressure_diff.fix()
+
+        @self.Constraint(self.flowsheet().time, doc="Saturation pressure constraint")
+        def outlet_pressure_sat(b, t):
+            return (
+                b.hot_side.properties_out[t].pressure + b.pressure_diff[t]
+                == b.hot_side.properties_out[t].pressure_sat
+            )
+
+    def initialize_build(self, *args, **kwargs):
+        """
+        Use the regular heat exchanger initialization, with the extraction rate
+        constraint deactivated; then it activates the constraint and calculates
```

Comment:
replace "extraction rate constraint" with appropriate constraints in description

Reply:
addressed